test/updateTimeToLive.js

```diff
@@ -0,0 +1,250 @@
+        var expiredItem = {
+          a: {S: sharedPk},
+          b: {S: helpers.randomString()},
+          c: {S: sharedGsiPk},
+          d: {S: helpers.randomString()},
+          TTL: {N: (timestampNow + 1).toString()},
+        }
+        var livingItem = {
+          a: {S: sharedPk},
+          b: {S: helpers.randomString()},
+          c: {S: sharedGsiPk},
+          d: {S: helpers.randomString()},
+          TTL: {N: (timestampNow + 1000000).toString()},
+        }
+        var batchWriteItemInput = {
+          RequestItems: {},
+        }
+        batchWriteItemInput.RequestItems[helpers.testRangeTable] = [
+          { PutRequest: { Item: expiredItem }},
+          { PutRequest: { Item: livingItem  }},
+        ]
+        request(helpers.opts('BatchWriteItem', batchWriteItemInput), function(err, res) {
+          if (err) return done(err)
+          res.statusCode.should.equal(200)
+
+          setTimeout(function(){
+            request(helpers.opts('Query', {
+              TableName: helpers.testRangeTable,
+              KeyConditionExpression: "a = :a",
+              ExpressionAttributeValues: {
+                ":a": expiredItem.a,
+              },
+            }), function(err, res) {
+              if (err) return done(err)
+              res.statusCode.should.equal(200)
+              res.body.Items.should.eql([livingItem], 'Expired item should be deleted from table')
+
+              request(helpers.opts('Query', {
+                TableName: helpers.testRangeTable,
+                IndexName: 'index1',
+                KeyConditionExpression: "a = :a",
+                ExpressionAttributeValues: {
+                  ":a": expiredItem.a,
+                },
+              }), function(err, res) {
+                if (err) return done(err)
+                res.statusCode.should.equal(200)
+                res.body.Items.should.eql([livingItem], "Expired Item should be deleted from LSI")
+
+                request(helpers.opts('Query', {
+                  TableName: helpers.testRangeTable,
+                  IndexName: 'index3',
+                  KeyConditionExpression: "c = :c",
+                  ExpressionAttributeValues: {
+                    ":c": expiredItem.c,
+                  },
+                }), function(err, res) {
+                  if (err) return done(err)
+                  res.statusCode.should.equal(200)
+                  res.body.Items.should.eql([livingItem], "Expired Item should be deleted from GSI")
+
+                  // teardown
+                  request(opts({
+                    TableName: helpers.testRangeTable,
+                    TimeToLiveSpecification: {AttributeName: "TTL", Enabled: false}
+                  }), function (err, res) {
+                    if (err) return done(err)
+                    res.statusCode.should.equal(200)
+                    done()
+                  })
+                })
+              })
+            })
+          }, 3000)
```

Comment:
Possible this could be reliably reduced? Those seconds add up in test suites!

Reply:
Replaced the fixed `setTimeout(..., 3000)` with `setInterval(... 50)` in https://github.com/architect/dynalite/pull/144/commits/669eea43353a653cd1efbb54b13a7036fc2667d1

On my machine, the test ends somewhere between 1900-2000 ms.